client.go

```diff
@@ -79,6 +126,9 @@ func NewClient(addr string, options ...func(c *Client) error) (*Client, error) {
 		timeout:    DefaultTimeout,
 		buf:        make([]byte, startBufSize),
 		maxBufSize: MaxParseTokenSize,
+		notify:     make(chan Notification, defaultNotificationBufSize),
+		err:        make(chan error),
+		disconnect: make(chan bool),
```

Comment:
idiomatic to use struct{} for this style of operation you can use close(c.disconnected) to signal the client has disconnected.

Reply:
I don't understand what I should change. How would I use close in a thread-safe way?

Reply:
close is always thread safe

Reply:
Wouldn't calling close twice cause a panic?

Reply:
close is always thread safe, to test its closed you can check in a select.